sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -121,13 +121,26 @@ func (k *listenerImpl) handleAllEvents() {
 	stopSignal := &k.stopSig
 
 	// Informers that need to be synced initially
-	handle(k.context, namespaceInformer, dispatchers.ForNamespaces(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
-	handle(k.context, secretInformer, dispatchers.ForSecrets(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
-	handle(k.context, saInformer, dispatchers.ForServiceAccounts(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
+	handle(k.context, namespaceInformer, dispatchers.ForNamespaces(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock, "namespace")
+	cache.WaitForNamedCacheSync("namespaces", stopSignal.Done(), namespaceInformer.HasSynced)
```

Comment:
@lvalerom does this way of waiting for the sync makes sense?

I guess something is wrong, as this lines blocks forever.

Reply:
For the record in case someone stumbles upon in the future: I think you need to call the `startAndWait` function to start the shared factories otherwise the informer won't start and won't sync